core/services/vrf/integration_test.go

```diff
@@ -141,6 +145,12 @@ func TestIntegration_VRF_WithBHS(t *testing.T) {
 	_ = createAndStartBHSJob(t, sendingKeys, app, cu.bhsContractAddress.String(),
 		cu.rootContractAddress.String(), "")
 
+	// Give buffer for log poller to get started.
+	gomega.NewGomegaWithT(t).Consistently(func() bool {
+		cu.backend.Commit()
+		return true
+	}, 5*time.Second, 1*time.Second).Should(gomega.BeTrue())
```

Comment:
Is the time buffer needed or is the # of finalized blocks that's the fix? If it's the latter you can just do

```suggestion
	// Give buffer for log poller to get started.
	for i := 0; i < 25; i++ {
		cu.backend.Commit()
	}
```

Reply:
It needs the time buffer for the log poller to start before the request gets emitted.

Reply:
Actually just updated the PR for an explicit log poller check, such that we don't have a race condition on it being ready before the request gets sent.